app/src/main/java/com/example/userinfo/MainActivity.kt

```diff
@@ -0,0 +1,148 @@
+package com.example.userinfo
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import androidx.core.view.isVisible
+import com.example.userinfo.databinding.ActivityMainBinding
+
+
+class MainActivity : AppCompatActivity() {
+    lateinit var binding: ActivityMainBinding
+    var visibleFieldsCheck: Boolean = false
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.validate.setOnClickListener {
+            if (fieldsValidationToCheckAllFieldsAreEntered() && emailValidation() && phoneNumberValidation() && pinCodeValidation()) {
+                binding.validUserName.setText(binding.userNameET.text.toString())
+                binding.validEmail.setText(binding.emailET.text.toString())
+                binding.validPhoneNumber.setText(binding.phoneNumberET.text.toString())
+                binding.validPinCode.setText(binding.pinCodeET.text.toString())
+                binding.validAddress.setText(binding.addressET.text.toString())
+                visibleFieldsCheck = true
+                fieldsDisplayOnValidate(visibleFieldsCheck)
+                binding.confirm.setOnClickListener {
+
+                    dataPassingFromActivityToActivityOnConfirm()
+                }
+                binding.cancel.setOnClickListener {
+                    visibleFieldsCheck = false
+                    fieldsDisplayOnValidate(visibleFieldsCheck)
+                }
+
+            }
+
+
+        }
+    }
+
+    private fun fieldsValidationToCheckAllFieldsAreEntered(): Boolean {
+        if ((binding.userNameET.text.toString().trim()
+                .isEmpty()) || (binding.emailET.text.toString().trim().isEmpty()) ||
+            (binding.phoneNumberET.text.toString().trim().isEmpty()) || (binding.pinCodeET.text.toString()
+                .trim().isEmpty()) ||
+            (binding.addressET.text.toString().trim().isEmpty())
+        ) {
+            Toast.makeText(this, "Every field is mandatory", Toast.LENGTH_LONG).show()
+            return false
```

Comment:
We do not have to return explicitily `false` here
you can return `!(binding.userNameET.text.toString().trim()........`